common/source/docs/common-uavlogviewer.rst

```diff
@@ -0,0 +1,90 @@
+
+Upon accessing, this panel is presented. You can "Open (a) Sample" file to experiment with, or load an ArduPilot .tlog or .bin  log file, either by drag and drop or by clicking to obtain a file explorer dialog to select the file.
+
+
+Main Screen
+===========
+
+Once a log is loaded, if it has GPS data, it will display this screen:
+
+.. image:: ../../../images/logviewer-3d-screen.png
+    :target: ../_images/logviewer-3d-screen.png
+
+which is the 3D flight viewer. You can replay the flight in 3D by pressing the "play" button, or "pause", "reverse", or  change playback "speed". Various map tools/options are in the upper right corner.
+
+In the upper left are setup menu buttons for the Plot (whose options are shown already), 3D panel, and additional widgets for parameters, log messages, and a joystick visualizer for the first four RC channels.
+
+3D View Options
+---------------
+
+
+.. image:: ../../../images/logviewer-3d-setup.png
+
+
+In this panel, you can change the camera mode in 3D playback, if waypoints and/or trajectory is shown, the size of the vehicle in the display, and the data source used for showing the trajectory.
+
+Plot Options
+------------
+
+.. image:: ../../../images/logviewer-plot-example1.png
+     :target: ../_images/logviewer-plot-example1.png
+
+This is one of the most powerful features of the viewer. This allows you to select which log values are plotted. You can search for an individual field to  plot, eg for CTUN.NavRoll you would type "Nav") or just navigate thru to the message with that field. Here we have plotted CTUN.NavRoll, CTUN.Roll,CTUN.NavPitch, and CTUN.Pitch.The Navigation controller's desired roll and pitch versu actual. ( For a complete list of log messages, see: `Copter Log Messages <https://ardupilot.org/copter/docs/logmessages.html>`__ , `Plane Log Messages <https://ardupilot.org/plane/docs/logmessages.html>`__ , or `Rover Log Messages <https://ardupilot.org/rover/docs/logmessages.html>`__ ).
+
+Presets
+-------
+
+Above the "Plot Individual Field" dropdown, is a "Presets" dropdown with many commonly used combinations of data plots already pre-configured.
+
+Plot Features
+-------------
+
+#. When each field is added, a Y axis number is assigned, and the data autoscaled and centered. This can be changed by hovering over the axis and scrolling to change scale, or clicking and moving the origin. If you wish to automatically align and scale several fields to one another, just set the y axis number to be the same. They will remain together from that point on.
+
+#. You can change the color code by clicking the color swatch.
+
+#. You can remove a data plot by clicking the trash can icon.
+
+#. You can even create and plot expressions that use the fields by "Add Expression" button. See ` Graph Expressions for MavExplorer <https://ardupilot.org/dev/docs/using-mavexplorer-for-log-analysis.html#graph-expressions>`__ for examples of using expressions in a plot analysis.
```

Comment:
I would add a note that currently only a subset of the "functions" available in mavproxy is available. I ported a bunch but there are still some missing

Reply:
if you have a list, I could include

Reply:
unfortunately I don't have it right now, I'll have to compare https://github.com/ArduPilot/pymavlink/blob/master/mavextra.py
and
https://github.com/ArduPilot/UAVLogViewer/blob/master/src/mavextra/mavextra.js
and also think of built-ins

Reply:
Can we skip this for now?

Reply:
sure....